### DISCUSSION TITLE
Another Created Date question

### DISCUSSION THREAD

**Discussion by BlankSlate56** [jonaswinkler/paperless-ng#593] [Support]
Now that I fixed my timezone, I've noticed that it has started picking up more of the dates correctly.  There is still a lot it is missing, but it is better.  One thing that I'm noticing is that it seems to struggle with numerical dates in the MM/DD/YYYY or MM-DD-YYYY numerical format.  It interprets alph-numeric dates such as January 11, 2020 correctly as 01/11/2020.

However, if a date is listed as 01/11/2020 on a document, paperless frequently transposes the month and the date.  So, instead of getting 01/11/2020, paperless is interpreting it as 11/01/2020.  This has been happening on probably half the dates that it is pulling in "correctly".  My hunch is that on dates it does pull in correctly that the "day" value is greater than 12 and so too big to be a month (01/17/2020), but I haven't confirmed that.

Am I missing a variable somewhere to tell it what date format to use?  I looked through the list, but maybe I missed it.

Thanks.

**Comment by jonaswinkler** [comment]
I haven't touched the date parsing logic at all (lots of complex regular expresssions, and it looks very intimidating), but I suspect the issue is that the date parsing module has no idea if it should parse 01/01/2020 as MM/DD/YYYY or as DD/MM/YYYY, and I don't know yet how to properly configure that.

I'll eventually touch the date parsing module of paperless and try to address some of the glaring issues (the code is not pretty and hard to read/change, and it's also VERY slow), but this is a pretty complicated thing to get right for all the different date formats out there.

**Reply by BlankSlate56** [reply]
Thanks for the reply.  Not trying to harass you about the date stuff by any means, I just hadn't seen similar issues in the log so I figured I'd put them out there.

Thanks for all your work on this.

**Reply by chris42** [reply]
> I haven't touched the date parsing logic at all (lots of complex regular expresssions, and it looks very intimidating), but I suspect the issue is that the date parsing module has no idea if it should parse 01/01/2020 as MM/DD/YYYY or as DD/MM/YYYY, and I don't know yet how to properly configure that.
> 
> I'll eventually touch the date parsing module of paperless and try to address some of the glaring issues (the code is not pretty and hard to read/change, and it's also VERY slow), but this is a pretty complicated thing to get right for all the different date formats out there.

If you are looking into the date functions, it might be worthwhile to think about some configuration options? Right now it seems, that it picks up the first date it finds in the document. Which in case of bank and credit card statements always fails on the: Ihre Abrechnung vom DD.MM.YYYY bis DD.MM.YYYY.

I guess perfect would be a context aware trigger word. E.g.: If you figured it is a bank statement, look for "bis" and pick the next date.

Sorry for the German mixup, but I would have no clue how a non German bank statement would look like.

**Comment by magnetic6** [comment]
Set this line in docker-compose.env to have the date parser use the month-day-year format.

PAPERLESS_DATE_ORDER=MDY
